### PR TITLE
Fix ID of custom dbal types

### PR DESCRIPTION
**Pull request by JustBlackBird**
This PR is a fix for #254  

Not sure if I didn't break anything but the tests are passing :smile:

### CODE REVIEW

**Comment by JustBlackBird** [comment]
It seems that something is wrong with HHVM and "nightly" test environments

**Comment by JustBlackBird** [comment]
Rebased

File: tests/bootstrap.php [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<?php
+
+require_once __DIR__ . '/../vendor/autoload.php';
+
+use Doctrine\DBAL\Types\Type;
+
+Type::addType('uuid', 'Doctrine\Tests\Common\DataFixtures\TestTypes\UuidType');
```
Comment (Majkl578) [inline tests/bootstrap.php]: IS there any particular reason why this is done here and not i.e. in test's `setUp`?

Comment (JustBlackBird) [reply tests/bootstrap.php]: It should be done only once per all test cases. It seems that I could move this to `setUpBeforeClass()` but not sure it's better

**Comment by JustBlackBird** [comment]
@Majkl578 I've moved type initialization to `setUpBeforeClass`. CI check is failed because it's unable to setup mongodb for php7.2 not because of the PR.

**Comment by JustBlackBird** [comment]
Is there anything else I should fix to get the PR merged?

**Comment by kisprof** [comment]
Hi there!

Whats the status of this PR? I just run into the same issue as in #254 and i was glad to see there is already a PR for it. Can i help in any way to speed up the merging process?

**Comment by JustBlackBird** [comment]
@Majkl578 ping

**Comment by JustBlackBird** [comment]
@Ocramius @Majkl578 
Is there anything I should fix to get the PR finally merged?

**Comment by JustBlackBird** [comment]
I've rebased changes to fix merge conflict.

**Comment by JustBlackBird** [comment]
Seems that `Doctrine\Tests\Common\DataFixtures\ORMPurgerTest::testGetAssociationTables` test case is already broken in master :(

**Comment by alcaeus** [comment]
I’ll take a look after the holidays. Thanks for following up.

**Comment by greg0ire** [comment]
I addressed the issue in #333

**Comment by JustBlackBird** [comment]
@greg0ire thank you!

Seems that #333 should be merged before this one

**Comment by greg0ire** [comment]
It just got merged :)

**Comment by greg0ire** [comment]
#333 is not merged up yet, hence why the build is not fixed. Are you sure you should target master though? Is there a BC-break in your PR?

**Comment by greg0ire** [comment]
The unit tests pass, but the coding standard do not.

File: tests/Doctrine/Tests/Common/DataFixtures/ProxyReferenceRepositoryTest.php [hunk 2]
```diff
@@ -113,6 +129,35 @@ public function testReferenceReconstruction()
         $this->assertInstanceOf(Proxy::class, $ref);
     }
 
+    public function testReconstructionOfCustomTypedId()
```
Comment (greg0ire) [inline tests/Doctrine/Tests/Common/DataFixtures/ProxyReferenceRepositoryTest.php]:
```suggestion
    public function testReconstructionOfCustomTypedId(): void
```

Comment (JustBlackBird) [reply tests/Doctrine/Tests/Common/DataFixtures/ProxyReferenceRepositoryTest.php]: Added `void` type hint

**Comment by JustBlackBird** [comment]
@greg0ire Thank you for help with codding style

**Comment by greg0ire** [comment]
I amended your commit with a message that sums up what I believe is a correct explanation for your PR. Please tell me if I got it wrong.

**Comment by JustBlackBird** [comment]
> I amended your commit with a message that sums up what I believe is a correct explanation for your PR. Please tell me if I got it wrong.

The message is fine, thank you!

**Comment by greg0ire** [comment]
Great, thanks @JustBlackBird !

**Comment by JustBlackBird** [comment]
@greg0ire thank you for getting the PR finally merged! :tada:

**Pull request merged**